lib/cloud-watch-event-listener.js

```diff
@@ -169,15 +258,25 @@ class CloudWatchEventListener extends events.EventEmitter {
         state,
         srid: srid ? srid : 'undefined',
         launched,
+        lastevent: generated,
       };
 
       if (workerType && provisionerId === this.provisionerId) {
         await this.state.upsertInstance(opts);
+        opts.metadataSource = 'api';
         log.info(opts, 'CloudWatch Event resulting in insertion');
       } else {
+        opts.metadataSource = 'api';
         log.debug(opts, 'Ignoring instance because it does not belong to this manager');
       }
     } else {
+      // For those events which are for instances which aren't being tracked in
+      // the database, we'll just blindly delete them.  This might have
+      // overhead on the database, but it does ensure that they're definitely
+      // removed from state.  The cost is likely a non-issue since the only way
+      // to avoid the removal would be the much more costly EC2 api calls to
+      // figure out metadata.  If we wanted to start tracking instance outcome,
+      // we could do it here (and also in the dbResponse truthy clause above).
```

Comment:
ah, maybe this comment here addresses my comment earlier about logging stats.  If we're putting guards around state and know what state an instance is, perhaps this is the best time to add those stats rather than waiting later.